app/models/street/StreetEdgeTable.scala

```diff
@@ -288,31 +288,68 @@ object StreetEdgeTable {
+  def getDistanceAudited(userId: UUID): Float = db.withSession { implicit session =>
 
-    val userStreets = selectAllStreetsAuditedByAUser(userId)
-    // get length of each street segment, sum the lengths, and convert from meters to miles
-    (userStreets.map(s => JTS.transform(s.geom, transform).getLength).sum * 0.000621371).toFloat
+    val dist = selectAllStreetsAuditedByAUserQuery(userId).groupBy(x => x).map(_._1.geom.transform(26918).length).list.sum
+    (dist * 0.000621371).toFloat // converts to miles
+  }
+
+  /** Returns the total distance audited by the specified user within the specified region, in miles */
+  def getDistanceAudited(userId: UUID, region: Int): Float = db.withSession { implicit session =>
+    // get the street edges from only this region
+    val auditedStreetsInRegion = for {
+      (_streetEdges, _streetEdgeRegions) <- selectAllStreetsAuditedByAUserQuery(userId).innerJoin(streetEdgeRegion).on(_.streetEdgeId === _.streetEdgeId)
+      if _streetEdgeRegions.regionId === region
+    } yield _streetEdges
+
+    // compute sum of lengths of the streets audited by the user in the region
+    val dist = auditedStreetsInRegion.groupBy(x => x).map(_._1.geom.transform(26918).length).list.sum
+    (dist * 0.000621371).toFloat // converts to miles
+  }
+
+  /** Returns the sum of the lengths of all streets in the region that have been audited */
+  def getDistanceAuditedInARegion(regionId: Int): Float = db.withSession { implicit session =>
+    val streetsInRegion = for {
+      (_streetEdges, _streetEdgeRegions) <- streetEdgesWithoutDeleted.innerJoin(streetEdgeRegion).on(_.streetEdgeId === _.streetEdgeId)
+      if _streetEdgeRegions.regionId === regionId
+    } yield _streetEdges
+
+    val auditedStreetsInARegion = for {
+      (_streetEdges, _auditTasks) <- streetsInRegion.innerJoin(completedAuditTasks).on(_.streetEdgeId === _.streetEdgeId)
+    } yield _streetEdges
+
+    // select distinct and sum the lengths of the streets
+    auditedStreetsInARegion.groupBy(x => x).map(_._1.geom.transform(26918).length).list.sum
+  }
+
+  /** Returns the sum of the lengths of all streets in the region */
+  def getTotalDistanceOfARegion(regionId: Int): Float = db.withSession { implicit session =>
+    val streetsInRegion = for {
+      (_streetEdges, _streetEdgeRegions) <- streetEdgesWithoutDeleted.innerJoin(streetEdgeRegion).on(_.streetEdgeId === _.streetEdgeId)
+      if _streetEdgeRegions.regionId === regionId
+    } yield _streetEdges
+
+    // select distinct and sum the lengths of the streets
+    streetsInRegion.groupBy(x => x).map(_._1.geom.transform(26918).length).list.sum
+  }
+
+  /** Returns the distance of the given street edge */
+  def getStreetEdgeDistance(streetEdgeId: Int): Float = db.withSession { implicit session =>
+    streetEdgesWithoutDeleted.filter(_.streetEdgeId === streetEdgeId).groupBy(x => x).map(_._1.geom.transform(26918).length).list.head
```

Comment:
These three functions are the ones that replace JTS.transform in RegionCompletionTable.scala